lib/site-inspector.rb

```diff
@@ -127,37 +134,213 @@ def to_json
     to_hash.to_json
   end
 
-  def to_hash
+  def http
+    details = {
+      # site-inspector's best guess
+      canonical: uri.to_s,
+      canonical_protocol: https? ? :https : :http,
+      canonical_endpoint: www? ? :www : :root,
+
+      endpoints: endpoints
+    }
+
+    combos = details[:endpoints]
+
+    details[:live] = (
+      combos[:https][:www][:live] or
+      combos[:https][:root][:live] or
+      combos[:http][:www][:live] or
+      combos[:http][:root][:live]
+    )
```

Comment:
A domain is considered "live" if at least one of its 4 combinations leads to a successful 200 response. This seems ripe for edge cases, but so is the normal mode's approach of only evaluating the canonical domain.